rust/main/hyperlane-base/src/contract_sync/cursors/sequence_aware/mod.rs

```diff
@@ -67,11 +70,17 @@ pub(crate) struct ForwardBackwardSequenceAwareSyncCursor<T> {
     forward: ForwardSequenceAwareSyncCursor<T>,
     backward: BackwardSequenceAwareSyncCursor<T>,
     last_direction: SyncDirection,
+    metrics: Arc<CursorMetrics>,
```

Comment:
curious what motivated storing the metrics in`ForwardBackwardSequenceAwareSyncCursor` instead of inside `ForwardSequenceAwareSyncCursor` and `BackwardSequenceAwareSyncCursor`. I think placing them inside the individual `Forward` and `Backward` cursors would be better for composability (say, a future cursor type that only uses the `ForwardSequenceAwareSyncCursor` - with the current setup you'd need to duplicate the metrics updating logic for it)

Reply:
It is a very good point. We have PR which makes the indexing direction configurable by chain. https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/4990

Reply:
Ah, was not aware of this type of functionality. Will have it updated. Thanks for the feedback